### ISSUE TITLE
[rllib] Checkpoints and Evaluation with TD3 

### ISSUE THREAD

**Issue by astronauti** [ray-project/ray#15062]
When saving the checkpoint or doing an evaluation step rllib with the tf2 framework produces an error. 

Ray version: 
TF: 2.4.1 
TFP: 0.12.1 
TFE: 2.4.0 
OS: linux-64 / centos/7.6.1810
#### Checkpointing
```
  File "test.py", line 62, in <module>
    checkpoint = agent.save(os.path.join(os.path.dirname(os.path.dirname(__file__)), 'Checkpoint'))
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/tune/trainable.py", line 321, in save
    checkpoint = self.save_checkpoint(checkpoint_dir)
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 690, in save_checkpoint
    pickle.dump(self.__getstate__(), open(checkpoint_path, "wb"))
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/agents/trainer_template.py", line 170, in __getstate__
    state = Trainer.__getstate__(self)
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 1228, in __getstate__
    state["worker"] = self.workers.local_worker().save()
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 1003, in save
    state = {
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 1004, in <dictcomp>
    pid: self.policy_map[pid].get_state()
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/policy/eager_tf_policy.py", line 540, in get_state
    state["_optimizer_variables"] = self._optimizer.variables()
AttributeError: 'NoneType' object has no attribute 'variables'
```

#### Evaluation
```
  File "test.py", line 84, in <module>
    result = agent.train()
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 529, in train
    raise e
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 515, in train
    result = Trainable.train(self)
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/tune/trainable.py", line 226, in train
    result = self.step()
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/agents/trainer_template.py", line 157, in step
    evaluation_metrics = self._evaluate()
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 749, in _evaluate
    self._sync_weights_to_workers(worker_set=self.evaluation_workers)
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 801, in _sync_weights_to_workers
    weights = ray.put(self.workers.local_worker().save())
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 1003, in save
    state = {
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 1004, in <dictcomp>
    pid: self.policy_map[pid].get_state()
  File "/opt/ohpc/pub/apps/conda/envs/ReinfLearning/lib/python3.8/site-packages/ray/rllib/policy/eager_tf_policy.py", line 540, in get_state
    state["_optimizer_variables"] = self._optimizer.variables()
AttributeError: 'NoneType' object has no attribute 'variables'
```

### Reproduction
```
import os, gym, ray 
from gym import spaces
import numpy as np

class dummyenv(gym.Env):
    metadata = {'render.modes': ['console']} # console
    def __init__(self, env_config):
        super(dummyenv, self).__init__()
        self.action_space = spaces.Box(low=np.array([0.1, 0]), high=np.array([1, 1]), dtype=np.float32) 
        self.observation_space = spaces.Box(low=0, high=255, shape=(2,), dtype=np.float32)
        observation = [0,0]

    def step(self, action):
        reward = 0.1
        info = {}
        observation= np.array([4, 3], dtype=np.float32)
        done = True
        return observation, reward, done, info

    def reset(self):
        observation = np.array([4, 3], dtype=np.float32)
        return observation  # reward, done, info can't be included

    def close(self):
          pass

config = {}
config['num_workers'] = 1 
config['evaluation_interval'] = 10
config['evaluation_num_episodes'] = 10
config['evaluation_config'] = {'explore': False}
config['evaluation_num_workers'] = 0 
config['num_gpus'] = 0 
config['framework'] = 'tf2'
config['eager_tracing'] = True
config['rollout_fragment_length'] = 1
config['train_batch_size'] = 1
config['normalize_actions'] = True

ray.init()

from ray.rllib.agents import ddpg
agent = ddpg.TD3Trainer(env=dummyenv, config=config)

for n in range(10):
    result = agent.train()
    checkpoint = agent.save(os.path.join(os.path.dirname(os.path.dirname(__file__)), 'Checkpoint'))
```

- [x] I have verified my script runs in a clean environment and reproduces the issue.

- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by astronauti** [comment]
@sven1977 @richardliaw

**Comment by mvindiola1** [comment]
@astronauti Will you check this with the nightly wheels?  This should already be fixed in this commit: #14119

**Comment by astronauti** [comment]
Thanks. With the nightly it is working 👍
